feature/navigation/drawer/src/main/kotlin/app/k9mail/feature/navigation/drawer/domain/entity/TreeFolder.kt

```diff
@@ -0,0 +1,96 @@
+package app.k9mail.feature.navigation.drawer.domain.entity
+
+import app.k9mail.core.mail.folder.api.Folder
+import app.k9mail.core.mail.folder.api.FolderType
+
+internal class TreeFolder(
+    var displayFolder: DisplayFolder? = null,
+) {
+    companion object {
+        fun createFromFolders(folders: List<DisplayFolder>, maxDepth: Int = 3): TreeFolder {
+            // Preparing root
+            val rootFolder = TreeFolder()
+
+            for (displayFolder in folders) {
+                // Managing exceptions
+                if (displayFolder is DisplayUnifiedFolder) {
+                    rootFolder.children.add(TreeFolder(displayFolder))
+                }
+                if (displayFolder !is DisplayAccountFolder) continue
+
+                // Inserting folder in tree
+                insertFolderInCurrentTree(rootFolder, displayFolder, maxDepth)
+            }
+
+            return rootFolder
+        }
+
+        private fun insertFolderInCurrentTree(
+            rootTree: TreeFolder,
+            displayFolder: DisplayAccountFolder,
+            maxDepth: Int,
+        ) {
+            val splittedFolderName = displayFolder.folder.name.split("/", limit = maxDepth + 1)
```

Comment:
There should be a fallback mechanism when folder names for not in the expected format.

Reply:
some IMAP server seems to use `.` as separator